frontend/src/pages/SignUpPage/ui/SignUpPage.tsx

```diff
@@ -1,30 +1,46 @@
-import { useForm } from "react-hook-form";
+import { Controller, useForm } from "react-hook-form";
+import { useNavigate } from "react-router";
 
 import { FetchBaseQueryError } from "@reduxjs/toolkit/query";
 
 import { useRegisterUserMutation } from "@/shared/api/generated";
-import { SrOnlyLabel } from "@/shared/styles/globalStyledComponents";
+import { AuthPage } from "@/shared/ui/AuthPage/AuthPage";
+import { PasswordInput } from "@/shared/ui/AuthPage/PasswordInput";
+import { TextInput } from "@/shared/ui/AuthPage/TextInput";
 import { Button } from "@/shared/ui/Button";
 import { zodResolver } from "@hookform/resolvers/zod";
 import { toast } from "sonner";
 import * as z from "zod";
 
 import * as S from "./SignUpPage.styles";
 
-const schema = z.object({
-  username: z.string().nonempty({ error: "Username cannot be empty." }),
-  email: z.email({ error: "Valid email address required." }),
-  password: z.string().nonempty({ error: "Password cannot be empty." }),
-});
+const schema = z
```

Comment:
since SignUpPage became bigger, I would suggest to move schema into a separate file close to SignUpPage component